content/specs/auth.md

```diff
@@ -0,0 +1,401 @@
+- `token_endpoint_auth_signing_alg` (string, optional): `none` is never allowed here. The current recommended and most-supported algorithm is `ES256`, but this may evolve over time. Authorization Servers will compare this against their supported algorithms.
+- `dpop_bound_access_tokens` (boolean, required): DPoP is mandatory for all clients, so this must be present and `true`
+- `jwks` (object with array of JWKs, optional): confidential clients must supply at least one public key in JWK format for use with JWT client authentication. Either this field or the `jwks_uri` field must be provided for confidential clients, but not both.
+- `jwks_uri` (string, optional): URL pointing to a JWKS JSON object. See `jwks` above for details.
+
+These fields are optional but recommended:
+
+- `client_name` (string, optional): human-readable name of the client
+- `client_uri` (string, optional): not to be confused with `client_id`, this is a homepage URL for the client. If provided, the `client_uri` must have the same hostname as `client_id`.
+- `logo_uri` (string, optional): URL to client logo. HTTPS is required over HTTP.
+- `tos_uri` (string, optional): URL to human-readable terms of service (ToS) for the client. HTTPS is required over HTTP.
+- `policy_uri` (string, optional): URL to human-readable privacy policy for the client. HTTPS is required over HTTP.
+
+See "OAuth Security Considerations" below for when `client_name`, `client_uri`, and `logo_uri` will or will not be displayed to end users.
+
+Additional optional client metadata fields are enumerated with [IANA](https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#client-metadata). Note that these are shared with the "Dynamic Client Registration" standard, which is not used directly by the atproto OAuth profile.
+
+#### Localhost Client Development
+
+When working on client software, it may be difficult for developers to publish in-progress client metadata at a public URL so that authorization servers can access it. This may even be true for development environments using a containerized Authorization Server and local DNS, because of SSRF protections against local IP ranges.
+
+To make development workflows easier, a special exception is made for clients with `client_id` having origin `http://localhost` (with no port number specified). Authorization Servers are encouraged to support this exception - including in production environments - but it is optional.
+
+In a localhost `client_id` scenario, the Authorization Server should verify that the scheme is `http`, and that the hostname is exactly `localhost` with no port specified. IP addresses (`127.0.0.1`, etc) are not supported. The path parameter must be empty (`/`).
+
+In the Authorization Request, the `redirect_uri` must match one of those supplied (or a default). Path components must match, but port numbers are not matched.
```

Comment:
```suggestion
In the Authorization Request, the `redirect_uri` must match one of those supplied (or a default). Path components must match, but port numbers are not matched.
```

From [OAuth2.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-10#authorization-request)
> "redirect_uri": OPTIONAL if only one redirect URI is registered for this client. REQUIRED if multiple redirict URIs are registered for this client. See [Section 2.3.2](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-10#multiple-redirect-uris).

Reply:
This entire section is about the loopback address case; re-iterating that this applies only for loopback addresses feels redundant?